zio-http/jvm/src/main/scala/zio/http/netty/NettyBody.scala

```diff
@@ -40,31 +40,46 @@ object NettyBody extends BodyEncoding {
   private[zio] def fromAsync(
     unsafeAsync: UnsafeAsync => Unit,
     knownContentLength: Option[Long],
-    contentTypeHeader: Option[Header.ContentType] = None,
-  ): Body = AsyncBody(
-    unsafeAsync,
-    knownContentLength,
-    contentTypeHeader.map(_.mediaType),
-    contentTypeHeader.flatMap(_.boundary),
-  )
+    contentTypeHeader: () => Option[Header.ContentType] = () => None,
```

Comment:
Note that allocation of `Function0` can be measured in benchmarks.

I would suggest the following:

1. Statically make `Header.ContentType` for common content types (e.g. `Header.ContentType.`application/json`)
2. Use fast comparison (or *maybe* hash map lookup) to decide which `Header.ContentType` to map to which standard Netty header. If Netty reuses the same char sequence, it could be even faster (`eq`).

If done properly this can as fast as `Function0` and will also be less disruptive to the code.